src/components/map/hooks/use-update-bottom-sheet-when-selected-entity-changes.tsx

```diff
@@ -33,17 +33,26 @@ export const useUpdateBottomSheetWhenSelectedEntityChanges = (
   mapSelectionAction: MapSelectionActionType | undefined,
   mapViewRef: RefObject<MapboxGL.MapView>,
   closeCallback: () => void,
-): Feature<Point, GeoJsonProperties> | undefined => {
+): {
+  selectedFeature: Feature<Point, GeoJsonProperties> | undefined;
+  onReportParkingViolation: () => void;
+} => {
   const isFocused = useIsFocused();
   const [selectedFeature, setSelectedFeature] = useState<Feature<Point>>();
   const {open: openBottomSheet, close: closeBottomSheet} = useBottomSheet();
   const analytics = useMapSelectionAnalytics();
   const navigation = useNavigation<RootNavigationProps>();
 
-  const closeWithCallback = () => {
+  const closeWithCallback = useCallback(() => {
     closeBottomSheet();
     closeCallback();
-  };
+  }, [closeBottomSheet, closeCallback]);
+
+  const onReportParkingViolation = useCallback(() => {
+    closeWithCallback();
+    analytics.logEvent('Mobility', 'Report parking violation clicked');
+    navigation.navigate('Root_ParkingViolationsSelectScreen');
+  }, [closeWithCallback, analytics, navigation]);
```

Comment:
I agree it would be better, but unfortunately since `onReportParkingViolation` depends on `closeWithCallback`, that won't work with the current setup. We should address this as well in the upcoming refactor I think.

```ts
const onReportParkingViolation = () => {...} // depends on closeWithCallback
const {
    ...,
    closeWithCallback,
  } = useMapSelectionChangeEffect(
    ...,
    onReportParkingViolation
  );
```

Reply:
Aha, ok 👍